functions.php

```diff
@@ -56,4 +56,59 @@ function legoCollection (array $legoSets) : string {
     }
 
     return $displayLego;
-}
\ No newline at end of file
+}
+
+/**
+ * Takes data retrieved from `post` and converts special characters
+ * @param array $postArray $_POST
+ * @return array cleansed array
+ */
+function cleanseData(array $postArray) : array {
+    $cleansedArray = [];
+    foreach ($postArray as $key => $postArrayItem) {
+        $cleansedArray[$key] =  htmlspecialchars($postArrayItem);
+    }
+    return $cleansedArray;
+}
+
+/**
+ * Checks if the item already exists in the database based on name
+ * @param array $itemArray cleansed array
+ * @param PDO $db database 'lily-collection'
+ * @return array empty array returned if item not found
+ */
+function checkItemExists(array $itemArray, PDO $db) : array {
+    $name = $itemArray['item-name'];
+
+    $findItem = $db->prepare("SELECT `item-name`,`image-URL`,`number-of-pieces`,`age-category`,`star-rating`,`buy-URL`,`retired` FROM `lego-sets` WHERE `item-name`= :itemName AND `deleted`=0");
+    $findItem->bindParam(':itemName',$name);
+    $findItem->execute();
+    $item=$findItem->fetch();
+
+    if($item == []) {
+        return [];
+    }
+    return $item;
+}
+
+/**
+ * Inserts data into database if it doesn't already exist
+ * @param array $itemArray cleansed array
+ * @param PDO $db database 'lily-collection'
+ */
+function insertToDatabase (array $itemArray, PDO $db) {
+
+    $name = $itemArray['item-name'];
```

Comment:
dont need to put these into variables, can just use them directly inside from the $itemArray inside the execute array